plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java

```diff
@@ -40,8 +71,114 @@ public TestResult execute(Map<String, String> testParameters) throws Exception {
     }
 
     @Override
+    @Nullable
     public ProcessHandler execute(TestExecutionContext context) {
+        String projectPath = context.getProjectPath();
+        IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectPath);
+        IJavaProject javaProject = JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+        if (javaProject.exists()) {
+            return startTestProcess(javaProject, context);
+        }
+
+        return null;
+    }
+
+    private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
+        File suiteFile = createSuite(context, javaProject);
+        if (suiteFile == null) {
+            throw new RuntimeException("Can't create TestNG suite xml file.");
+        }
+
+        JavaParameters parameters = new JavaParameters();
+        parameters.setJavaExecutable("java");
+        parameters.setMainClassName("org.testng.CheTestNGLauncher");
+        parameters.setWorkingDirectory("/tmp");
```

Comment:
I see your point. 
So my solution is: 

- Working directory is the project root directory; 
- The directory where the reports will be generated will created in target folder (project output directory) and named test-output if project output directory (for example target) doesn't exist test-output will be created in the project root.

You can see this changes with my last commits.

@davidfestal @vparfonov Is it OK?